config/initializers/rack_attack.rb

```diff
@@ -0,0 +1,23 @@
+Rack::Attack.cache.store = ActiveSupport::Cache::MemoryStore.new
+Rack::Attack.throttled_response = ->(env) { [429, {}, [ActionView::Base.new.render(file: 'public/429.html')]] }
+
+ActiveSupport::Notifications.subscribe('rack.attack') do |name, start, finish, request_id, req|
+  Airbrake.notify Exception.new(message: "#{req.ip} has been rate limited for url #{req.url}", data: req.env['rack.attack.throttle_data'])
```

Comment:
This looks great. 
Only thing that I feel needs to change is the MemoryStore usage - seeing as we're seeing memory problems.

Can you use ActiveSupport::Cache::MemCacheStore in production?   You might even be able to use the cache store defined in the environment.